framework/decode/dx12_replay_consumer_base.cpp

```diff
@@ -5879,6 +5883,20 @@ bool Dx12ReplayConsumerBase::CopyResourceAsyncQueue(const std::vector<format::Ha
             }
         }
 
+        // TODO: Some resource states are not avaiable for compute type queue. This error could cause devie lost,
+        // so remove them. It might have more unavailable states that should be removed.But this's not a good idea.
+        // A better way might be that using a new direct type queue to copy resources. But it has two issues.
+        // 1: If the resource is swapchain, it has to use its queue.
+        // 2: It has to deal with sync issue between the new queue and the original queue.
```

Comment:
I am leaning towards using a new direct type queue to copy resources. For the potential issues:
1. When a swapchain resource is detected, use the swapchain's queue. Otherwise use the direct type queue.
2. The sync issue should be solvable without too much modification to the current sync framework for dump resources.